### PR TITLE
Effect Dust OK

### PR DESCRIPTION
**Pull request by hensldm**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Tru/z_en_tru.c [hunk 1]
```diff
@@ -81,7 +81,7 @@ static Color_RGBA8 D_80A8B25C[] = {
 static f32 D_80A8B274[] = { 60.0f, 255.0f, 60.0f };
 
 static TexturePtr D_80A8B280[] = {
-    gDust8Tex, gDust7Tex, gDust6Tex, gDust5Tex, gDust4Tex, gDust3Tex, gDust2Tex, gDust1Tex,
+    gEffDust8Tex, gEffDust7Tex, gEffDust6Tex, gEffDust5Tex, gEffDust4Tex, gEffDust3Tex, gEffDust2Tex, gEffDust1Tex,
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Tru/z_en_tru.c:84]: Wow there are a lot of copies of this!

File: src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.c [hunk 2]
```diff
@@ -1,30 +1,192 @@
...
+
+    if ((this->life <= this->rLifespan) && (this->life >= (this->rLifespan - 7))) {
+        if (this->rLifespan >= 5) {
+            this->rTexIndex = this->rLifespan - this->life;
+        } else {
+            this->rTexIndex = ((this->rLifespan - this->life) * (8 / this->rLifespan));
+        }
+    } else {
+        this->rTexIndex = 7;
+    }
```
Comment (EllipticEllipsis) [inline src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.c:146]: I think a lot of these 7s and 8s can sensibly be written in terms of `ARRAY_COUNT()`?

File: src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.h [hunk 3]
```diff
@@ -3,6 +3,16 @@
 
 #include "global.h"
 
+#define DUST_DRAWFLAG0 (0)
```
Comment (EllipticEllipsis) [inline src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.h:6]: I don't think we usually make defines for flags with value 0.

Comment (hensldm) [reply src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.h:6]: Yes, though this time it actually has a meaning besides just NONE. It has its own type of drawing.

Comment (EllipticEllipsis) [reply src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.h:6]: To me it seems more like it's just not doing the things that the others do (although of course it's treating them more like an enum than actual flags since it's if-elses).

Comment (hensldm) [reply src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.h:6]: The main reason I added it was for the function calls where the flags are just 4 (rand color offset), since I think it is clearer to or with this flag since the bottom three bits do act like an enum (even if they aren't for reasons).

File: src/overlays/actors/ovl_En_Fg/z_en_fg.c [hunk 4]
```diff
@@ -445,7 +445,7 @@ void EnFg_UpdateDust(EnFgEffectDust* dustEffect) {
 }
 
 TexturePtr sDustTex[] = {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Fg/z_en_fg.c:447]:
```suggestion
TexturePtr sDustTextures[] = {
```

File: assets/xml/objects/gameplay_keep.xml [hunk 5]
```diff
@@ -1416,14 +1416,14 @@
...
-        <Texture Name="gDust1Tex" OutName="dust_1" Format="i8" Width="32" Height="32" Offset="0x8DBE0" />
-        <Texture Name="gDust2Tex" OutName="dust_2" Format="i8" Width="32" Height="32" Offset="0x8DFE0" />
-        <Texture Name="gDust3Tex" OutName="dust_3" Format="i8" Width="32" Height="32" Offset="0x8E3E0" />
-        <Texture Name="gDust4Tex" OutName="dust_4" Format="i8" Width="32" Height="32" Offset="0x8E7E0" />
-        <Texture Name="gDust5Tex" OutName="dust_5" Format="i8" Width="32" Height="32" Offset="0x8EBE0" />
-        <Texture Name="gDust6Tex" OutName="dust_6" Format="i8" Width="32" Height="32" Offset="0x8EFE0" />
-        <Texture Name="gDust7Tex" OutName="dust_7" Format="i8" Width="32" Height="32" Offset="0x8F3E0" />
-        <Texture Name="gDust8Tex" OutName="dust_8" Format="i8" Width="32" Height="32" Offset="0x8F7E0" />
+        <Texture Name="gEffDust1Tex" OutName="dust_1" Format="i8" Width="32" Height="32" Offset="0x8DBE0" />
+        <Texture Name="gEffDust2Tex" OutName="dust_2" Format="i8" Width="32" Height="32" Offset="0x8DFE0" />
```
Comment (engineer124) [inline assets/xml/objects/gameplay_keep.xml:1420]:
```suggestion
        <Texture Name="gEffDust1Tex" OutName="eff_dust_1" Format="i8" Width="32" Height="32" Offset="0x8DBE0" />
        <Texture Name="gEffDust2Tex" OutName="eff_dust_2" Format="i8" Width="32" Height="32" Offset="0x8DFE0" />
```
For the remaining ones too

File: src/code/z_effect_soft_sprite_old_init.c [hunk 6]
```diff
@@ -91,63 +91,76 @@ void EffectSsDust_Spawn(PlayState* play, u16 drawFlags, Vec3f* pos, Vec3f* veloc
 
 void func_800B0DE0(PlayState* play, Vec3f* pos, Vec3f* velocity, Vec3f* accel, Color_RGBA8* primColor,
                    Color_RGBA8* envColor, s16 scale, s16 scaleStep) {
-    EffectSsDust_Spawn(play, 0, pos, velocity, accel, primColor, envColor, scale, scaleStep, 10, 0);
+    EffectSsDust_Spawn(play, DUST_DRAWFLAG0, pos, velocity, accel, primColor, envColor, scale, scaleStep, 10,
+                       DUST_UPDATE_NORMAL);
 }
```
Comment (engineer124) [inline src/code/z_effect_soft_sprite_old_init.c:96]: Not part of this PR, but do you have any ideas how we could name these 13 variations of dust spawn functions? `func_800B0DE0` to `func_800B12F0`?

Comment (hensldm) [reply src/code/z_effect_soft_sprite_old_init.c:96]: Not yet, it is going to need to come with the further doc pass of dust later. Mainly need to figure out what each drawflag actually does, then decided names based on that and the other slight differences.

File: src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.h (hunk 3 above)
Comment (hensldm) [reply src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.h:6]: For more context here, I did try spawning dust that runs threw each drawflags intended branch (0, 1, 2). 0,2 look very similar if not identical (though could be a bad graphics pluggin), while 1 definitely looks different. Personally I still like `DRAWFLAG_0` since it still acts like an enum and has its own code path, even if it doesn't do much. Perhaps I don't call them flags but do something like 
```c
#define Dust_Draw0 0
#define Dust_Draw1 1
#define Dust_Draw2 2
```
since they aren't really flags, but also really an enum.

**Review by engineer124 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]
